crates/interfaces/src/p2p/error.rs

```diff
@@ -34,6 +34,11 @@ impl RequestError {
     pub fn is_retryable(&self) -> bool {
         matches!(self, RequestError::Timeout | RequestError::ConnectionDropped)
     }
+
+    /// Whether the error happened because the channel was closed.
+    pub fn is_channel_closed(&self) -> bool {
+        matches!(self, RequestError::ChannelClosed)
+    }
```

Comment:
@onbjerg this might affect retry logic, because now `RequestError::ChannelClosed` is propagated to the client if a session was closed while handling the request.

but there's another channel from the `FetchClient -> Network` which is still open

so `ChannelClosed` could be retryable, but not necessarily, it'd be cleaner to handle this separately somehow.

should convert `ChannelClosed` to `ConnectionDropped` I think?

Reply:
okay I solved this by changing what's called when the session is closed

https://github.com/paradigmxyz/reth/blob/09cbf96719cb332b332dcbd576060b867d33cec2/crates/net/network/src/state.rs#L385-L399

now all requests are answered with `ConnectionDropped`